dsl_survey/atlas_stack_language/overview.yaml

```diff
@@ -0,0 +1,91 @@
+title: Atlas Stack Language
+
+abbreviation: ASL
+
+copyright: Netflix, Inc.
+
+license: Apache License V2
+
+isOSS: true
+
+paradigms:
+  - stack
+  - declarative
+
+supportedTelemetryTypes:
+  - type: metrics
+    callouts:
+      - title: Consolidation
+        description: >
+          Atlas processes and stores data at pre-configured and fixed intervals rather
+          than at individual timestamps. All queries must use the default or include
+          a valid "step" size that determines the interval window of each value returned
+          from a query. The majority of data is stored at a 60 second step.
+          
+          Wider query time ranges automatically increase the step size to reduce the
+          number of data points in a graph or output. Users can choose custom step
+          sizes at regular multiples of the base step size. The consolidation function
+          applied to values differs based on the data type stored. E.g. sums for counts,
+          average for rates, max for gauges.
+      - title: Default Aggregation
+        description: >
+          Data matching the predicate is aggregated via `sum` by default. Groupings must
+          be applied to drill down into individual time series.
+
+inspirations:
+  - RRDTool
+
+description: >
+  An RPN-style stack based DSL for the open source Atlas metrics system from Netflix. 
+  The design was meant to ease transition from an RRDtool based system to a more 
+  flexible, dimension based distributed store. A rich syntax allows for simple single 
+  line filtering, grouping and aggregation of metrics while also supporting complex 
+  data analysis. The DSL includes presentation aspects as the API is capable of 
+  generating PNG graph images for storing and sharing similar to the previous RRDtool.
+
+designCallouts:
```

Comment:
Some other callouts that could be interesting:

- Backend vs streaming queries. Most ASL with a few exceptions like offset and filters can be executed against a backend store or in a streaming context. This allows us to schedule most queries where we want for things like alerts and the user typically doesn't need to worry about which context it is running in.
- For ASL we categorize the operations, roughly into choosing, aggregation, math, presentation (CAMP). Some contexts like queries on metadata to project the set of keys or values for a given key just support the choosing category. For data queries choosing/aggregation can be pushed to the data layer and be distributed similar to basic map/reduce operations. The math operations are also broken up into basic, stateful, and filters depending on what is needed for an operations. For example, stateful operations keep history and would put restrictions on the execution order of the expression if partitioning computation across time.